### PR TITLE
Stop referencing packages that became inbox on modern .NET.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
See #65088.

@jmarolf, @jaredpar, @RikkiGibson

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
CI is green.

File: src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (jjonescz) [inline src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj:1]: File encoding changed?

File: src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj [hunk 2]
```diff
@@ -26,18 +26,20 @@
...
-    <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
     <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
     <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="$(SystemRuntimeCompilerServicesUnsafeVersion)" />
-    <PackageReference Include="System.Threading.Tasks.Extensions" Version="$(SystemThreadingTasksExtensionsVersion)" />
     <PackageReference Include="System.Text.Encoding.CodePages" Version="$(SystemTextEncodingCodePagesVersion)" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
+    <PackageReference Include="System.Threading.Tasks.Extensions" Version="$(SystemThreadingTasksExtensionsVersion)" />
+  </ItemGroup>
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj:42]: Please use `PackageReference ... Condition=` vs creating a new `ItemGroup`

**Review by jaredpar — CHANGES_REQUESTED** [review]

File: src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj (hunk 1 above)
Comment (teo-tsirpanis) [reply src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj]: The BOM was removed, brought it back.

File: src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj (hunk 2 above)
Comment (teo-tsirpanis) [reply src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj]: Done.

**Review by jmarolf — APPROVED** [review]
I sign off on the IDE changes

**Review by jaredpar — APPROVED** [review]

**Comment by teo-tsirpanis** [comment]
Conflicts were resolved. Can it be merged?

**Review by 333fred — APPROVED** [review]

**Pull request merged**